tmt/hardware.py

```diff
@@ -1335,7 +1335,7 @@ def _parse_generic_spec(spec: Spec) -> BaseConstraint:
         group.constraints += [_parse_location(spec)]
 
     if 'system' in spec:
-        group.constraints += [_parse_system(spec)]
+        group.constraints += [_parse_system(spec['system'])]
```

Comment:
Could you include a fix for `location` as well? Check line 1335, it suffers from the same issue :(